src/server.c

```diff
@@ -1778,6 +1778,23 @@ static void sendGetackToReplicas(void) {
     argv[1] = shared.getack;
     argv[2] = shared.special_asterick; /* Not used argument. */
     replicationFeedSlaves(server.slaves, -1, argv, 3);
+
+    if (server.io_threads_num == 1) return;
+
+    /* Since we want to read the ACK ASAP we make sure to return any replicas
```

Comment:
Can we generalize (and rename) `putSlavesNeedingAckReadInPendingClientsToIOThreads()` with a `flags` argument (say whether to also check `slaveFromIOThreadNeedsAckRead()` before sending to io threads) and use it here too?